client/client_service.go

```diff
@@ -45,7 +45,9 @@ func (c *Client) Put(endpoint string, obj models.Model) (*container.Container, e
 		return nil, err
 	}
 
+	c.Mutex.Lock()
```

Comment:
The parallelism issue is only on PATCH to the best of my knowledge. We might only need the mutex on PATCH. Did you see it also in non-patch requests?

Reply:
It shouldn't be an issue with concurrent POST or PUT requests, but I think having a concurrent POST and PATCH can cause issues as well, which is why I safeguarded all write operations.

Reply:
Will keep the mutex for PUT for now as it is minimal impact on provider.